backend/models/courseunit.mgmodel.ts

```diff
@@ -0,0 +1,99 @@
+import mongoose, { Schema, Document } from "mongoose";
+import { PageType } from "../types/courseTypes";
+
+export type ElementSkeleton = {
+  id: string;
+  x: number;
+  y: number;
+  w: number;
+  h: number;
+  content: string;
+};
+
+const ElementSkeletonSchema: Schema = new Schema({
+  x: {
+    type: Number,
+    required: true,
+  },
+  y: {
+    type: Number,
+    required: true,
+  },
+  w: {
+    type: Number,
+    required: true,
+  },
+  h: {
+    type: Number,
+    required: true,
+  },
+  content: {
+    type: String,
```

Comment:
Content type could be an enum as there should be a fixed number of types of components. Ie "Text Box" | " Match Box" | "Fill in the Blank"

However, maybe as the number of types of components increase in the future this becomes more tedious / messy.

Reply:
The idea with the `content` property of the `ElementSkeleton` is that it just contains an ID which maps us to a separate collection of `CourseElements` - which is why we're making it a `String`.

The `courseElements` schema is still a WIP haha, but 100% agree we should have an enum for the different types of components when we get there :)

Reply:
also LMAO love the papas pizzeria pic